evm/src/witness/transition.rs

```diff
@@ -227,11 +227,25 @@ fn get_op_special_length(op: Operation) -> Option<usize> {
     }
 }
 
+// These operations might trigger a stack overflow, typically those pushing without popping.
+// Kernel-only pushing instructions aren't considered; they can't overflow.
+fn might_overflow_op(op: Operation) -> bool {
+    match op {
+        Operation::Pc | Operation::Push(0) => true,
+        Operation::Push(_) => true,
+        // SWAP doesn't, but share a flag with DUP.
+        Operation::Dup(_) | Operation::Swap(_) => true,
+        // Doesn't directly push, but the syscall it's returning from might.
+        Operation::ExitKernel => true,
+        _ => false,
+    }
+}
```

Comment:
How do we ensure that this code here does not go out of sync with the `MIGHT_OVERFLOW` constant?

Reply:
Ah, we don't, that's right. I'll turn it into a function like `get_op_special_length` or `fill_op_flag`.